lib/nrf_modem_lib/nrf_modem_lib.c

```diff
@@ -192,6 +203,52 @@ int nrf_modem_lib_shutdown(void)
 	return 0;
 }
 
+/**
+ * Modem library fault handler.
+ *
+ * If a different error handling is required, the handler can be defined in the application
+ * by setting the NRF_MODEM_LIB_ON_FAULT_APPLICATION_SPECIFIC Kconfig option.
+ *
+ * @param[in] fault_info Modem fault information. Contain the fault reason, and,
+ *                       in some cases, the modem program counter.
+ */
+
+#if defined CONFIG_NRF_MODEM_LIB_ON_FAULT_DO_NOTHING
+void nrf_modem_fault_handler(struct nrf_modem_fault_info *fault_info)
+{
+	LOG_ERR("Modem error: 0x%x, PC: 0x%x", fault_info->reason, fault_info->program_counter);
+}
```

Comment:
Seems that modem_shell build is failing here with:
CONFIG_NRF_MODEM_LIB_TRACE_ENABLED=y
CONFIG_NRF_MODEM_LIB_LOG_LEVEL_OFF=y

The latter ones triggers the problem in compilation:
/home/jani/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: modules/nrf/lib/nrf_modem_lib/lib..__nrf__lib__nrf_modem_lib.a(nrf_modem_lib.c.obj): in function `nrf_modem_fault_handler':
/home/jani/ncs/nrf/lib/nrf_modem_lib/nrf_modem_lib.c:220: undefined reference to `log_dynamic_nrf_modem'

Seen in CI as well:
https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-7571/21/

->

https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-nrf-iot_mosh/job/master/1062/consoleFull

Reply:
@nordic-krch is this a problem with the logger not being able to compile the log macro from a C file that uses `LOG_MODULE_DECLARE` if the other C file with `LOG_MODULE_REGISTER` has the level set to `_OFF` ?

Reply:
Thanks, did not know that `_DECLARED` could also specify the level.